### PR TITLE
Dynamic config preview

### PR DESCRIPTION
**Pull request by annabaker**
# Description
Agreed upon GA content for Dynamic Config (previously known as setup workflows).

### CODE REVIEW

File: jekyll/_cci2/configuration-cookbook.md [hunk 1]
```diff
@@ -566,6 +566,171 @@ workflows:
...
+## Dynamic Configuration
+{: #dynamic-configuration }
+
+This section provides various examples in using CircleCI's Dynamic Configuration feature. To get started, and for more
+information, see the [documentation]({{ site.baseurl }}/2.0/dynamic-config).
+
+### A Basic Example
+{: #a-basic-example }
+
+The following is a basic example using CircleCI's Dynamic Configuration feature. In this example, we assume that a
```
Comment (circleci-docs) [inline jekyll/_cci2/configuration-cookbook.md:578]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/configuration-cookbook.md [hunk 2]
```diff
@@ -566,6 +566,171 @@ workflows:
...
+  - The `build-service-1` job uses the `maven` orb to compile/install the service1 code, but skips any tests
+  - The `build-service-2` job uses the `maven` orb to compile/install the service2 code, but skips any tests
+  - The `run-integration-tests` job uses the `maven` orb to run any integration tests
+- Define four workflows, three of which are conditionally executed:
+  - The `service-1` workflow triggers the `build-service-1` job when the pipeline parameter value mapped to run-build-service-1-job is set to `true`
+  - The `service-2` workflow triggers the `build-service-2` job when the pipeline parameter value mapped to run-build-service-2-job is set to `true`
+  - The `run-integration-tests` workflow will run if the `run-build-service-1-job` or `run-build-service-2-job` pipeline parameters have been updated to `true` based on the results of the `path-filtering` orb
+  - The `check-updated-files` workflow will always run any time this pipeline is triggered
+
+See the `path-filtering` [orb documentation](https://circleci.com/developer/orbs/orb/circleci/path-filtering) for more
```
Comment (circleci-docs) [inline jekyll/_cci2/configuration-cookbook.md:731]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/dynamic-config.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
+---
+layout: classic-docs
+title: "Dynamic Configuration"
+description: "Docs page on using Setup Workflows for Dynamic Configuration"
+version:
+- Cloud
+---
+
+You may find that instead of manually creating each and every individual CircleCI configuration per project,
+you would prefer to generate these configurations dynamically depending on specific [pipeline]({{ site.baseurl }}/2.0/concepts/#pipelines)
```
Comment (circleci-docs) [inline jekyll/_cci2/dynamic-config.md:10]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/dynamic-config.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+version:
+- Cloud
+---
+
+You may find that instead of manually creating each and every individual CircleCI configuration per project,
+you would prefer to generate these configurations dynamically depending on specific [pipeline]({{ site.baseurl }}/2.0/concepts/#pipelines) 
+parameters or file-paths.
+
+This becomes particularly useful in cases where your team is using a monorepo, or a single repository, as opposed to
+using multiple repositories to store your code. In the case of using a monorepo, it is of course optimal to only trigger
```
Comment (circleci-docs) [inline jekyll/_cci2/dynamic-config.md:14]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/dynamic-config.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+- Cloud
+---
+
+You may find that instead of manually creating each and every individual CircleCI configuration per project,
+you would prefer to generate these configurations dynamically depending on specific [pipeline]({{ site.baseurl }}/2.0/concepts/#pipelines) 
+parameters or file-paths.
+
+This becomes particularly useful in cases where your team is using a monorepo, or a single repository, as opposed to
+using multiple repositories to store your code. In the case of using a monorepo, it is of course optimal to only trigger 
+specific builds in specific areas of your project. Otherwise, all of your microservices/project will go through
```
Comment (circleci-docs) [inline jekyll/_cci2/dynamic-config.md:15]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/dynamic-config.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+---
+
+You may find that instead of manually creating each and every individual CircleCI configuration per project,
+you would prefer to generate these configurations dynamically depending on specific [pipeline]({{ site.baseurl }}/2.0/concepts/#pipelines) 
+parameters or file-paths.
+
+This becomes particularly useful in cases where your team is using a monorepo, or a single repository, as opposed to
+using multiple repositories to store your code. In the case of using a monorepo, it is of course optimal to only trigger 
+specific builds in specific areas of your project. Otherwise, all of your microservices/project will go through 
+the entirety of your build, test, and deployment processes when any single update is introduced.
```
Comment (circleci-docs) [inline jekyll/_cci2/dynamic-config.md:16]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/dynamic-config.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+These computed pipeline parameters and/or generated `config.yaml` files can then be passed into an additional `config.yaml`
+that potentially exist in outside directories.
+
+In summary, CircleCI's Dynamic Configuration allows you to:
+
+- Execute conditional workflows/commands
+- Pass pipeline parameter values and/or generate additional configuration
+- Trigger separate `config.yml` configurations which exist outside the default parent `.circleci/` directory
+
+To use our Dynamic Configuration feature, you can add the key `setup` with a value of `true` to the top-level of your
```
Comment (circleci-docs) [inline jekyll/_cci2/dynamic-config.md:32]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/dynamic-config.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+that potentially exist in outside directories.
+
+In summary, CircleCI's Dynamic Configuration allows you to:
+
+- Execute conditional workflows/commands
+- Pass pipeline parameter values and/or generate additional configuration
+- Trigger separate `config.yml` configurations which exist outside the default parent `.circleci/` directory
+
+To use our Dynamic Configuration feature, you can add the key `setup` with a value of `true` to the top-level of your 
+parent configuration file (in the `.circleci/` directory). This will designate that configuration as a `setup workflow`
```
Comment (circleci-docs) [inline jekyll/_cci2/dynamic-config.md:33]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/dynamic-config.md [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+- Execute conditional workflows/commands
+- Pass pipeline parameter values and/or generate additional configuration
+- Trigger separate `config.yml` configurations which exist outside the default parent `.circleci/` directory
+
+To use our Dynamic Configuration feature, you can add the key `setup` with a value of `true` to the top-level of your 
+parent configuration file (in the `.circleci/` directory). This will designate that configuration as a `setup workflow` 
+configuration.
+
+See the [Getting Started](#getting-started-with-dynamic-config-in-circleci) section below for more
```
Comment (circleci-docs) [inline jekyll/_cci2/dynamic-config.md:36]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/dynamic-config.md [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+parent configuration file (in the `.circleci/` directory). This will designate that configuration as a `setup workflow` 
+configuration.
+
+See the [Getting Started](#getting-started-with-dynamic-config-in-circleci) section below for more 
+information.
+
+## Getting Started with Dynamic Config in CircleCI
+{: #getting-started-with-dynamic-config-in-circleci }
+
+To get started with Dynamic Config in CircleCI:
```
Comment (circleci-docs) [inline jekyll/_cci2/dynamic-config.md:42]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/dynamic-config.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+- Select the project you are interested in, in the **Projects** dashboard from the CircleCI application.
+- Click the **Project Settings** button in the upper-right corner.
+- On the left-hand panel, select **Advanced**.
+- Towards the bottom, toggle the switch for **Run Setup Workflows** to the "on" position (it should be blue)
+
+<!-- INCLUDE A SCREENSHOT AFTER GA -->
+
+Now, your project has the ability to dynamically generate configuration.
+
+When using Dynamic Configuration, at the end of the `setup workflow`, a `continue` job from the [`continuation`](https://circleci.com/developer/orbs/orb/circleci/continuation)
```
Comment (circleci-docs) [inline jekyll/_cci2/dynamic-config.md:53]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/dynamic-config.md [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+
+## Dynamic Configuration FAQs
+{: #dynamic-config-faqs }
+
+### Pipeline Parameters
+{: #pipeline-parameters }
+
+**Q:** I thought pipeline parameters could only be used with the API?
+
+**A:** Previously, this was true. But using our Dynamic Configuration feature, you can set pipeline parameters dynamically,
```
Comment (circleci-docs) [inline jekyll/_cci2/dynamic-config.md:71]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/configuration-cookbook.md [hunk 13]
```diff
@@ -566,6 +566,180 @@ workflows:
...
+
+The following examples of dynamic configuration usage are provided below:
+
+- [A basic example]({{ site.baseurl }}/2.0/configuration-cookbook/?section=examples-and-guides#a-basic-example)
+- [Execute specific `workflows` or `steps` based on which files are modified]({{ site.baseurl }}/2.0/configuration-cookbook/?section=examples-and-guides#execute-specific-workflows-or-steps-based-on-which-files-are-modified)
+
+### A basic example
+{: #a-basic-example }
+
+The following is a basic example using CircleCI's dynamic configuration feature. In this example, we assume that a
```
Comment (circleci-docs) [inline jekyll/_cci2/configuration-cookbook.md:583]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

**Review by Aytac998 — APPROVED** [review]

File: jekyll/_cci2/dynamic-config.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+## Getting started with dynamic config in CircleCI
+{: #getting-started-with-dynamic-config-in-circleci }
+
+To get started with Dynamic Config in CircleCI: 
+
+- Select the project you are interested in, in the **Projects** dashboard from the CircleCI application.
+- Click the **Project Settings** button in the upper-right corner.
+- On the left-hand panel, select **Advanced**.
+- Towards the bottom, toggle the switch for **Enable dynamic config using setup workflows** to the "on" position, as shown below:
+
```
Comment (circleci-docs) [inline jekyll/_cci2/dynamic-config.md:48]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 2

File: jekyll/_cci2/dynamic-config.md [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+- Click the **Project Settings** button in the upper-right corner.
+- On the left-hand panel, select **Advanced**.
+- Towards the bottom, toggle the switch for **Enable dynamic config using setup workflows** to the "on" position, as shown below:
+  
+
+![Enable dynamic config in the UI]({{ site.baseurl }}/assets/img/docs/dynamic-config-enable.png)
+
+Now, your project has the ability to dynamically generate and update configuration.
+
+When using dynamic configuration, at the end of the `setup workflow`, a `continue` job from the [`continuation`](https://circleci.com/developer/orbs/orb/circleci/continuation)
```
Comment (circleci-docs) [inline jekyll/_cci2/dynamic-config.md:54]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/dynamic-config.md [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+
+## Dynamic configuration FAQs
+{: #dynamic-config-faqs }
+
+### Pipeline parameters
+{: #pipeline-parameters }
+
+**Q:** I thought pipeline parameters could only be used with the API?
+
+**A:** Previously, this was true. But using our dynamic configuration feature, you can set pipeline parameters dynamically,
```
Comment (circleci-docs) [inline jekyll/_cci2/dynamic-config.md:72]: MD009/no-trailing-spaces Trailing spaces Expected: 0 or 2; Actual: 1

File: jekyll/_cci2/configuration-cookbook.md [hunk 17]
```diff
@@ -566,6 +566,180 @@ workflows:
...
+
+- Add the line `setup: true` to the top-level of our config, to designate it for use of CircleCI's dynamic configuration feature
+- Invoke the `continuation` orb so we can use it.
+- Define a job called `setup` that uses the `continuation` orb as an [`executor`]({{ site.baseurl }}/2.0/executor-intro/). This job:
+    - Calls the [`checkout`]({{ site.baseurl }}/2.0/configuration-reference/#checkout) step to checkout code from the configured repository.
+    - Calls the [`run`]({{ site.baseurl }}/2.0/configuration-reference/#checkout) step to execute the existing `generate-config` script, so we can pass its output to the `continue` job of the `continuation` orb.
+    - Continues running the pipeline based on what configuration is provided to the required `configuration_path`.
+- Lastly, we call the `setup` job defined above as a part of our `workflow`
+
+For a more in-depth explanation of what the `continuation` orb does, see the orb's source code in the
```
Comment (thomasr8) [inline jekyll/_cci2/configuration-cookbook.md:628]:
```suggestion
For a more in-depth explanation of what the `continuation` orb does, review the orb's source code in the
```

File: jekyll/_cci2/configuration-cookbook.md [hunk 18]
```diff
@@ -566,6 +566,180 @@ workflows:
...
+[CircleCI Developer Hub](https://circleci.com/developer/orbs/orb/circleci/continuation?version=0.1.2) or see the
```
Comment (thomasr8) [inline jekyll/_cci2/configuration-cookbook.md:629]:
```suggestion
[CircleCI Developer Hub](https://circleci.com/developer/orbs/orb/circleci/continuation?version=0.1.2) or review the
```

File: jekyll/_cci2/configuration-cookbook.md [hunk 19]
```diff
@@ -566,6 +566,180 @@ workflows:
...
+[Dynamic configuration FAQ]({{ site.baseurl }}/2.0/dynamic-config#dynamic-config-faqs).
+
+### Execute specific `workflows` or `steps` based on which files are modified
+{: #execute-specific-workflows-or-steps-based-on-which-files-are-modified }
+
+You may find that you would like to conditionally run a `workflow` or `step` based upon changes made to a specific fileset.
+This would be beneficial in the case of your code/microservices being stored in a monorepo, or a single repository.
+
+To achieve this, CircleCI has provided the [`path-filtering`](https://circleci.com/developer/orbs/orb/circleci/path-filtering)
+orb, which allows a pipeline to continue execution based upon the specific paths of updated files.
```
Comment (thomasr8) [inline jekyll/_cci2/configuration-cookbook.md:639]:
```suggestion
orb, which enables a pipeline to continue execution based upon the specific paths of updated files.
```

File: jekyll/_cci2/configuration-cookbook.md [hunk 20]
```diff
@@ -566,6 +566,180 @@ workflows:
...
+
+For example, consider a monorepo structure like so:
```
Comment (thomasr8) [inline jekyll/_cci2/configuration-cookbook.md:641]:
```suggestion
For example, consider a monorepo structure like the example shown below:
```

File: jekyll/_cci2/configuration-cookbook.md [hunk 21]
```diff
@@ -566,6 +566,180 @@ workflows:
...
+# our defined jobs
+jobs:
+  check-updated-files:
+    - path-filtering/filter:
+        # 3-column, whitespace-delimited mapping. One mapping per line: <regex path-to-test> <parameter-to-set> <value-of-pipeline-parameter>.
+        mapping: |
+          service1/.* run-service-1-job true
+          service2/.* run-service-2-job true
+        base-revision: master
+        # this is the path of the configuration we should trigger once path filtering and pipeline parameter value updates are complete. in this case, we are using the parent dynamic configuration itself.
```
Comment (thomasr8) [inline jekyll/_cci2/configuration-cookbook.md:686]:
```suggestion
        # this is the path of the configuration we should trigger once path filtering and pipeline parameter value updates are complete. In this case, we are using the parent dynamic configuration itself.
```

File: jekyll/_cci2/configuration-cookbook.md [hunk 22]
```diff
@@ -566,6 +566,180 @@ workflows:
...
+    jobs:
+      - run-integration-tests
+  always-run:
+    jobs:
+      - check-updated-files
+```
+
+In the above configuration, we:
+
+- Add the line `setup: true` to the top-level of our config, to designate it for use of CircleCI's dynamic configuration feature
```
Comment (thomasr8) [inline jekyll/_cci2/configuration-cookbook.md:726]:
```suggestion
- Add the line `setup: true` to the top-level of our config, to designate it for use of CircleCI's dynamic configuration feature.
```

File: jekyll/_cci2/dynamic-config.md [hunk 23]
```diff
@@ -0,0 +1,90 @@
+---
+layout: classic-docs
+title: "Dynamic Configuration"
+description: "Docs page on using Setup Workflows for Dynamic Configuration"
+version:
+- Cloud
+---
+
+You may find that instead of manually creating each and every individual CircleCI configuration per project,
+you would prefer to generate these configurations dynamically depending on specific [pipeline]({{ site.baseurl }}/2.0/concepts/#pipelines)
```
Comment (thomasr8) [inline jekyll/_cci2/dynamic-config.md:10]:
```suggestion
you would prefer to generate these configurations dynamically, depending on specific [pipeline]({{ site.baseurl }}/2.0/concepts/#pipelines)
```

File: jekyll/_cci2/dynamic-config.md [hunk 24]
```diff
@@ -0,0 +1,90 @@
...
+This becomes particularly useful in cases where your team is using a monorepo, or a single repository, as opposed to
+using multiple repositories to store your code. In the case of using a monorepo, it is of course optimal to only trigger
+specific builds in specific areas of your project. Otherwise, all of your microservices/sub-projects will go through 
+the entirety of your build, test, and deployment processes when any single update is introduced.
+
+In both of these (and many other) use cases, automatic, dynamic generation of your configuration files will optimize your
+CircleCI experience and save your team both time and money.
+
+CircleCI's dynamic configuration feature uses a `setup workflow` configuration. A `setup workflow` can contain jobs that
+`setup` children pipelines through computed pipeline parameters, or by generating followup pipelines via pre-existing scripts.
```
Comment (thomasr8) [inline jekyll/_cci2/dynamic-config.md:22]:
```suggestion
`setup` children pipelines through computed pipeline parameters, or by generating follow-up pipelines via pre-existing scripts.
```

File: jekyll/_cci2/dynamic-config.md [hunk 25]
```diff
@@ -0,0 +1,90 @@
...
+To get started with Dynamic Config in CircleCI:
+
+- Select the project you are interested in, in the **Projects** dashboard from the CircleCI application.
+- Click the **Project Settings** button in the upper-right corner.
+- On the left-hand panel, select **Advanced**.
+- Towards the bottom, toggle the switch for **Enable dynamic config using setup workflows** to the "on" position, as shown below:
+
+![Enable dynamic config in the UI]({{ site.baseurl }}/assets/img/docs/dynamic-config-enable.png)
+
+Now, your project has the ability to dynamically generate and update configuration.
```
Comment (thomasr8) [inline jekyll/_cci2/dynamic-config.md:51]:
```suggestion
Now, your project has the ability to dynamically generate and update a configuration.
```

**Review by thomasr8 — APPROVED** [review]
Just some minor comments.....feel free to incorporate them if you wish.

**Review by thomasr8 — APPROVED** [review]

**Pull request merged**